spikeinterface/toolkit/qualitymetrics/pca_metrics.py

```diff
@@ -298,21 +301,152 @@ def nearest_neighbors_isolation(all_pcs, all_labels, this_unit_id, max_spikes_fo
+    
+    nn_isolation = np.min(isolation)
+    return nn_isolation
+
+def nearest_neighbors_noise_overlap(waveform_extractor: si.WaveformExtractor, 
+                                    this_unit_id: int, max_spikes_for_nn: int=1000,
+                                    n_neighbors: int=5, n_components: int=10, seed: int=0):
+    """Calculates unit noise overlap based on NearestNeighbors search in PCA space.
+
+    Based on noise overlap metric described in Chung et al. (2017) Neuron 95: 1381-1394.
+
+    Rough logic:
+    ------------
+    1) Generate a noise cluster by randomly sampling voltage snippets from recording.
+    2) Subtract projection onto the weighted average of noise snippets
+       of both the target and noise clusters to correct for bias in sampling.
+    3) Compute the isolation score between the noise cluster and the target cluster.
+    
+    Implementation details:
+    -----------------------
+    As with nn_isolation, the clusters that are compared (target and noise clusters)
+    have the same number of spikes.
+    
+    See docstring for `_compute_isolation` for the definition of isolation score.
+    
+    Parameters:
+    -----------
+    we: si.WaveformExtractor
+    this_unit_id: int
+        ID of unit for which this metric will be calculated
+    max_spikes_for_nn: int
+        max number of spikes to use per cluster
+    n_neighbors: int
+        number of neighbors to check membership of
+    n_components: int
+        number of PC components to project the snippets
+    seed: int
+        seed for random subsampling of spikes
+
+    Outputs:
+    --------
+    nn_noise_overlap : float
+    """
+
+    # set random seed
+    rng = np.random.default_rng(seed=seed)
+    
+    # get random snippets from the recording to create a noise cluster
+    recording = waveform_extractor.recording
+    noise_cluster = get_random_data_chunks(recording, return_scaled=True,
+                                           num_chunks_per_segment=max_spikes_for_nn,
+                                           chunk_size=waveform_extractor.nsamples, seed=seed)
+    noise_cluster = np.reshape(noise_cluster, (max_spikes_for_nn, waveform_extractor.nsamples, -1))
+    
+    # get waveforms for target cluster
+    waveforms = waveform_extractor.get_waveforms(unit_id=this_unit_id)
+    
+    # adjust the size of the target and noise clusters to be equal
+    if waveforms.shape[0] > max_spikes_for_nn:
+        wf_ind = rng.choice(waveforms.shape[0], max_spikes_for_nn, replace=False)
+        waveforms = waveforms[wf_ind]
+        n_snippets = max_spikes_for_nn
+    elif waveforms.shape[0] < max_spikes_for_nn:
+        noise_ind = rng.choice(noise_cluster.shape[0], waveforms.shape[0], replace=False)
+        noise_cluster = noise_cluster[noise_ind]
+        n_snippets = waveforms.shape[0]
+    else:
+        n_snippets = max_spikes_for_nn
+
+    # restrict to channels with significant signal
+    mean_waveform = np.mean(waveforms, axis=0)
```

Comment:
would it be more robust to take median instead? Or, leave it to the user as an extra arg (see [here](https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/waveform_extractor.py#L354))

```suggestion
    # 'mode' can be either 'average' or 'median'
    mean_waveform = waveform_extractor.get_template(unit_id=this_unit_id, mode=mode)
```

Reply:
Agree that median would be better. Also replaced with call to `get_template`. I would prefer to just use median not have it as an extra arg because it would be yet another arg for the user to learn about and decide on, and is in any case not very important because the only reason we do this is to identify the max channel and sample.